src/Microsoft.AspNet.Routing/DependencyInjection/RoutingServiceCollectionExtensions.cs

```diff
@@ -11,24 +11,27 @@ namespace Microsoft.Extensions.DependencyInjection
     /// <summary>
     /// Contains extension methods to <see cref="IServiceCollection"/>.
     /// </summary>
-    public static class ServiceCollectionExtensions
+    public static class RoutingServiceCollectionExtensions
     {
-        /// <summary>
-        /// Configures a set of <see cref="RouteOptions"/> for the application.
-        /// </summary>
-        /// <param name="services">The services available in the application.</param>
-        /// <param name="setupAction">An action to configure the <see cref="RouteOptions"/>.</param>
-        public static void ConfigureRouting(
+        public static IServiceCollection AddRouting(this IServiceCollection services)
+        {
+            return AddRouting(services, null);
+        }
+
+        public static IServiceCollection AddRouting(
             this IServiceCollection services,
-            Action<RouteOptions> setupAction)
+            Action<RouteOptions> configureOptions)
         {
-            if (setupAction == null)
+            services.AddOptions();
+            services.TryAddTransient<IInlineConstraintResolver, DefaultInlineConstraintResolver>();
+            services.TryAddSingleton(UrlEncoder.Default);
+
+            if (configureOptions != null)
             {
-                throw new ArgumentNullException(nameof(setupAction));
+                services.Configure(configureOptions);
             }
 
-            services.TryAddSingleton(UrlEncoder.Default);
-            services.Configure(setupAction);
```

Comment:
This should have been added in `AddRouting`, and there shouldn't be a `ConfigureRouting`